Adds smolder module to extras.

mcameron opened the pull request:
Similar to the uri module in core.
More powerful, with advanced testing capabilities.
See https://github.com/sky-shiny/smolder

sivel commented:
Some initial feedback based on automated scanning of this module:

```
============================================================================
smolder.py
============================================================================
ERROR: version_added be should 2.0. Currently 1.9
ERROR: ansible.module_utils.basic import not near main()
ERROR: GPLv3 license header not found
ERROR: Interpreter line is not "#!/usr/bin/python"
WARNING: No RETURN provided
```

mcameron commented:
Hmm.  Happy to resolve these issues.  However 
`Interpreter line is not "#!/usr/bin/python"`
mans that this module won't work inside of a virtualenv.  Surely that's not right.

bcoca commented:
@mcameron that is what ansible_python_interpreter is for, which will not work with your current shebang

mcameron commented:
Done.

sivel commented:
I have a few pieces of additional feedback:
1. Your first 2 examples are missing indentation under `smolder:`
2. Your examples, are using `guthub`, instead of `github`.  Might be worth changing those so that they work out of the box.
3. You seem to be adding reset color codes to the stdout, regardless or whether `outcomes['colour_output']` is set or not.

mcameron commented:
Thanks for the feedback.  I've fixed 1 and 2.  

Regarding resetting the colour of output, the outcomes['colout_output'] flag controls colour display within a smolder test, and the reset is in there to remove colours set by ansible.  Imo, the affect is quite pleasing  and logical.

I have run fixup and all changes are in one commit now.

mcameron commented:
Thoughts?

gregdek commented:
@sivel any comment? Can we get a "shipit"? :)

robynbergeron commented:
Or since it's a new module, get a +1 each to passing guidelines and working :)

mscherer commented on network/smolder.py:
```diff
@@ -0,0 +1,126 @@
+
+    test_obj = Charcoal(test=test_input, host=module.params["host"])
+    total_failed_tests += test_obj.failed
+    total_passed_tests += test_obj.passed
```
Since there is no loop, why do you do that ? (I mean, you do not need a specific intermediate variable)

mcameron commented:
Made eight more commits, fixed up and rebased. Currently supporting this file in two repos, so will be relieved when we finally get this approved.  That being said, if there is any more feedback, I will be very pleased to take it on board.

mcameron commented:
@robynbergeron done. :)

gregdek commented:
Thanks @mcameron for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

gregdek commented:
@mcameron A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

mcameron commented:
Extinguished.
